### PR TITLE
Allow create and remove functions to operate properly when app_id is …

### PR DESCRIPTION
**Pull request by jaleksynas**
…present and you are using a technical role account rather than admin. Also removed some code duplication.
#1895

### CODE REVIEW

**Comment by rds13** [comment]
👍

**Comment by ahmedgharib** [comment]
I assume this will work with app manager as well?

**Comment by jaleksynas** [comment]
The main change (other than some code cleanup and de-duplication) is to use app specific endpoints for tester create/delete.  If a user has the appropriate permissions for the operation in the web UI, this allows the same using pilot.

File: pilot/lib/pilot/manager.rb [hunk 1]
```diff
@@ -56,5 +56,18 @@ def fetch_app_identifier
       UI.verbose("App identifier (#{result})")
       return result
     end
+
+    # Perform the app_id lookup based solely on the passed parameters
+    # nil if app_id is not configured.
+    def find_app_id_no_prompt
+      return @apple_id if @apple_id
+      app_filter = (config[:apple_id] || config[:app_identifier])
+      if app_filter
```
Comment (KrauseFx) [inline pilot/lib/pilot/manager.rb]: This should probably be 
```ruby
return nil unless app_filter
```

**Comment by mpirri** [comment]
This looks awesome, @jaleksynas -- can you add some test coverage to this code? Otherwise, we're excited to get this functionality into fastlane!

**Comment by ohwutup** [comment]
closing this in favor of https://github.com/fastlane/fastlane/pull/5145

**Pull request closed without merging**